### ISSUE TITLE
Unfolding thread only unfolds first status

### ISSUE THREAD

**Issue by monsterpit-daggertooth** [glitch-soc/mastodon#446]
*(our fork is caught up to commit f4ed382)*

When using the `Show more for all` feature, only the first status' CW is unfolded.
Tested in Firefox and Chrome.

![screen shot 2018-04-26 at 12 08 07](https://user-images.githubusercontent.com/30352165/39320790-cd4d4c66-494a-11e8-95e7-a51bae9461a7.png)

* * * *

- [X] I searched or browsed the repo’s other issues to ensure this is not a duplicate.

**Comment by ThibG** [comment]
I can reproduce it, and I'm pretty sure it worked some time ago, so something may have been wrong when rebasing the changes… having a look…

**Comment by ThibG** [comment]
Update: `componentWillReceiveProps` doesn't seem to be called at all, neither from the “Show more for all” feature nor from enabling/disabling collapsed toots.
This is most likely a change in React.JS lifecycles either due to upgrading React.JS a short while ago or making use of the new `getSnapshotBeforeUpdate` lifecycle method. Investigating further…

**Comment by ThibG** [comment]
This is indeed caused by defining `getSnapshotBeforeUpdate` although this isn't clearly documented in React.JS…
The correct thing to do is probably to switch to the new `getDerivedStateFromProps` lifecycle, as `componentWillReceiveProps` is considered unsafe/obsolete anyway, but that new method don't let us inspect the old props, so switching over isn't completely straightforward.

**Comment by ThibG** [comment]
I'm afraid I really don't get how React.JS work. I attempted translating the current `componentWillReceiveProps` to `getDerivedStateFromProps` and while I *think* I got it right… returning a state change does not trigger a rendering with this new state… things like pressing a key in the composer will cause a re-rendering. But I'm extremely confused as to why it doesn't render `getDerivedStateFromProps`'s result immediately.

TL;DR: I can't React.JS, please help

**Comment by ThibG** [comment]
Turned out to be a bug in `react-redux` or one of its dependencies. Updating it to 5.0.7 fixed rendering not being triggered as a result of property changes. I'll clean up my mess and come with a proper fix shortly.

**Comment by ThibG** [comment]
This has been fixed
